### ISSUE TITLE
Add db/dt algorithm. 

### ISSUE THREAD

**Issue by hasimpson-usgs** [usgs/geomag-algorithms#115]

**Comment by jmfee-usgs** [comment]
Looks like numpy already supports this, algorithm process method would call diff on all traces in a stream:

https://docs.scipy.org/doc/numpy/reference/generated/numpy.diff.html

**Comment by erigler-usgs** [comment]
Obspy has a `differentiate` function for Traces too. It uses NumPy's `gradient` function instead of `diff`. I believe `diff` is more consistent with how we calculated dB/dt in the past, but frankly I think ObsPy's approach is better.

Perhaps more important: we should take care that dB/dt units are correct. I'm not sure what we did in the past, but we *should* use units of 1/second, whether calculating dB/dt for minute, second, or even 10Hz data, at least if it is to ever be used as more than a simple diagnostic.

We should talk about this, and possibly assign this to a student.
